sdk/core/Azure.Core/src/Pipeline/Internal/RedirectPolicy.cs

```
@@ -168,5 +173,77 @@ public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePol
+                {
+                    return false;
+                }
+            }
+
+            parsedValue = uri;
+            return true;
+        }
+
+        // The normal client header parser just casts bytes to chars (see GetString).
+        // Check if those bytes were actually utf-8 instead of ASCII.
+        // If not, just return the input value.
+        private static string DecodeUtf8FromString(string input)
+        {
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                return input;
+            }
+
+            bool possibleUtf8 = false;
+            for (int i = 0; i < input.Length; i++)
+            {
+                if (input[i] > (char)255)
+                {
+                    return input; // This couldn't have come from the wire, someone assigned it directly.
+                }
+                else if (input[i] > (char)127)
+                {
+                    possibleUtf8 = true;
+                    break;
+                }
+            }
+            if (possibleUtf8)
+            {
+                byte[] rawBytes = new byte[input.Length];
+                for (int i = 0; i < input.Length; i++)
+                {
+                    if (input[i] > (char)255)
+                    {
+                        return input; // This couldn't have come from the wire, someone assigned it directly.
+                    }
+                    rawBytes[i] = (byte)input[i];
+                }
+                try
+                {
+                    // We don't want '?' replacement characters, just fail.
+                    System.Text.Encoding decoder = System.Text.Encoding.GetEncoding("utf-8", System.Text.EncoderFallback.ExceptionFallback,
+                        System.Text.DecoderFallback.ExceptionFallback);
+                    return decoder.GetString(rawBytes, 0, rawBytes.Length);
+                }
+                catch (ArgumentException) { } // Not actually Utf-8
```

Are you worried about swallowing this exception?

---

Nope, we expect the `GetString` to throw if it fails to decode.